extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js

```diff
@@ -993,14 +1000,21 @@ export class AmpLightboxGallery extends AMP.BaseElement {
   // TODO (cathyxz): make this generalizable to more than just images
   enter_() {
     const sourceElement = this.getCurrentElement_().sourceElement;
-    return this.shouldAnimate_(sourceElement)
-        .then(shouldAnimate => {
-          if (shouldAnimate) {
-            return this.transitionIn_(sourceElement);
-          } else {
-            return this.fade_(0, 1);
-          }
-        });
+    if (!this.elementTypeCanBeAnimated_(sourceElement)) {
+      return this.fade_(0, 1);
+    }
+
+    return Promise.all([
+      this.shouldAnimate_(sourceElement),
+      this.getCurrentElement_().imageViewer.signals()
+          .whenSignal(CommonSignals.LOAD_END),
+    ]).then(values => {
+      if (values[0]) {
```

Comment:
nit: For `Promise.all` values, assign the items from the array to a named variable so the code is easier to scan